barman/clients/walrestore.py

```diff
@@ -243,15 +244,14 @@ def try_deliver_from_spool(config, dest_file):
     :param argparse.Namespace config: the configuration from command line
     :param dest_file: The destination file object
     """
-    spool_file = os.path.join(config.spool_dir, config.wal_name)
+    spool_file = Path(config.spool_dir, config.wal_name)
```

Comment:
Do you have a specific reason for using `Pathlib` instead of `os.path` here?

We are planning to do this and other changes in the whole codebase at once, so if is not necessary for your patch we would like to avoid introducing this kind of changes right now.

I hope that his makes sense to you.

Reply:
I prefer `Pathlib` and in that case used it to work around typing-issues. Reverted that change.

I also found another issues I addressed in this commit.

Reply:
I do prefer `Pathlib` too, but barman is has a really old codebase that we kept compatible with pyhton 2 for a long time. Said so, we really want to bring in some "modern" changes into the barman code soon :)